ansible/README.md

```diff
@@ -0,0 +1,190 @@
+# Ansible playbook for Tendermint applications
+
+![Ansible plus Tendermint](img/a_plus_t.png)
+
+* [Prerequisites](#Prerequisites)
+* [Ansible setup](#Ansible setup)
+* [Running the playbook](#Running the playbook)
+
+The playbooks in this folder run [ansible](http://www.ansible.com/) roles which:
+
+* install and configure basecoin or ethermint
+* start/stop basecoin or ethermint and reset their configuration
+
+## Prerequisites
+
+* Ansible 2.0 or higher
+* SSH key to the servers
+
+Optional for DigitalOcean droplets:
+* DigitalOcean API Token
+* python dopy package
+
+Head over to the [Terraform folder](https://github.com/tendermint/tools/tree/master/terraform-digitalocean) for a description on how to get a DigitalOcean API Token.
+
+Optional for Amazon AWS instances:
+* Amazon AWS API access key ID and secret access key.
+
+The cloud inventory scripts come from the ansible team at their [GitHub](https://github.com/ansible/ansible) page. You can get the latest version from the contrib/inventory folder.
+
+## Ansible setup
+
+Ansible requires a "command machine" or "local machine" or "orchestrator machine" to run on. This can be your laptop or any machine that can run ansible. (It does not have to be part of the cloud network that hosts your servers.)
+
+Use the official [Ansible installation guide](http://docs.ansible.com/ansible/intro_installation.html) to install Ansible. Here are a few examples on basic installation commands:
+
+Ubuntu/Debian:
+```
+sudo apt-get install ansible
+```
+
+CentOS/RedHat:
+```
+sudo yum install epel-release
+sudo yum install ansible
+```
+
+Mac OSX:
+```
+sudo easy_install pip
+sudo pip install ansible
+```
+
+To make life easier, you can start an SSH Agent and load your SSH key(s). This way ansible will have an uninterrupted way of connecting to your servers.
+
+```
+ssh-agent > ~/.ssh/ssh.env
+source ~/.ssh/ssh.env
+
+ssh-add private.key
```

Comment:
Hm..., let me rephrase that once more, I think Apple is missing a point here. The ssh-agent's job is to cache your private key in memory for security and convenience purposes. The workflow is the following:
1. During bootup you provide a passphrase manually so the private key can be read and decrypted from the file system into memory.
2. You make sure your environment is pointing to the ssh-agent using environment variables so if any processes (ssh for example) needs authenticating, they can request the ssh-agent to sign stuff that the process presents. This way the private key is secured with encryption on the file system but still available for signing through ssh-agent. Note that the ssh-agent will not provide the private key to the process, but instead it will receive bytes from the process, sign them with the private key and send it back to the process. So the private key is also safe in memory. (As long as the kernel is doing it's job properly.)
This is more convenient than manually entering a passphrase every time SSH (or a similar process) requires authentication, as well as more secure because the process doesn't get direct access to the private key.

Then comes Apple and cuts through this. The point of the workflow above was that in security there is always a manual point somewhere that initiates the whole security algorithm (consensus for symmetric secret) but ssh-agent allows you to go through this only once (per reboot cycle) and have the convenience of better security and less interaction for the system. Now Apple adds a feature (`-K`) which is not compatible with any other Unix or Linux system but allows to connect to yet another Apple-only system (keychain), to store the passphrase centrally and sync it to a cloud service - managed by Apple. So, until now you had to hack the in-memory ssh-agent to read the private key from memory, now you have multiple attack vectors: you can hack the keychain, the Apple cloud service or the ssh-agent memory. Also, since you still have to run ssh-agent at every reboot, you can try to hack into the ssh-agent - keychain communication. (I didn't check, but it would be interesting to see how secure that is.) By adding another layer of abstraction they made something quite secure into a nightmare. All parts of your ssh private key is stored by a digital system now.
Note that if you think about it, it is more secure to store your private key as plain text without encryption on your encrypted Apple hard-drive because that will not end up in the cloud. (Well, except if you turned on the feature to upload your folders to iCloud drive.) I know this is exaggeration, but it has its points.
Admittedly linux has features like HashiCorp Vault, to store multiple secrets in one place, but there at least you manage all the data and not a third-party with dubious license agreements.
So, everything summarized, ssh-agent/ssh-add requires you to enter the passphrase at each boot for a reason and I'm against using `-K` in OSX.

With that said, I've added a note to the readme so people know about this option for OSX.